### DISCUSSION TITLE
API Documentation

### DISCUSSION THREAD

**Discussion by OneYellowLemon** [Jikoo/OpenInv#71] [Q&A]
Hello! Is there any API documentation/javadocs? How would I go about opening ender chests via the API in my own GUI?

**Comment by OneYellowLemon** [comment]
@Jikoo Are you able to point me in the right direction?

**Comment by Jikoo** [comment]
Every API method has attached documentation. If you want to open OI's ender chests, [obtain a Player instance for the offline player](https://github.com/Jikoo/OpenInv/blob/3103e8822b77d2292c526266f623d42a045d908a/api/src/main/java/com/lishid/openinv/IOpenInv.java#L139), then [obtain an ISpecialEnderChest](https://github.com/Jikoo/OpenInv/blob/3103e8822b77d2292c526266f623d42a045d908a/api/src/main/java/com/lishid/openinv/IOpenInv.java#L112), then [open that for the target player](https://github.com/Jikoo/OpenInv/blob/3103e8822b77d2292c526266f623d42a045d908a/api/src/main/java/com/lishid/openinv/IOpenInv.java#L221).

**Reply by Jikoo** [reply]
If you want to see a functional example, you could look at [how OI opens inventories/ender chests](https://github.com/Jikoo/OpenInv/blob/3103e8822b77d2292c526266f623d42a045d908a/plugin/src/main/java/com/lishid/openinv/commands/OpenInvCommand.java).
